src/header.rs

```diff
@@ -146,11 +146,12 @@ pub(crate) fn default_commit_types() -> Vec<CommitMessage> {
 
 pub fn validate(
     spec: Vec<CommitMessage>,
+    allow_caps_type: bool,
     commit_type: &str,
     scope: &str,
     description: &str,
 ) -> Result<bool, Error> {
-    let commit_type = spec.iter().find(|x| x.commit_type == commit_type);
+    let commit_type = spec.iter().find(|x| x.commit_type == commit_type || (allow_caps_type && x.commit_type.to_ascii_uppercase() == commit_type));
```

Comment:
To be case insensitive, the commit type has to be uppercase as well, right? 
Can you maybe also add tests for that scenario?

Reply:
I added test cases further down in this file, which cases do you think are missing?

Reply:
yeah, but also that one is verifying that case insensitivity is not achieved. 

Let's take the following example:

```rust

#[test]
fn test_validate_allow_caps_types_more_cases() {
    let commit_types = vec![
        CommitMessage {
            commit_type: "FEAT".to_string(),
            required: vec!["scope".to_string(), "description".to_string()],
        },
        CommitMessage {
            commit_type: "build".to_string(),
            required: vec!["description".to_string()],
        },
    ];
    let test_cases = vec![
        ("feat", "scope", "description"),
        ("FEAT", "scope", "description"),
        ("build", "", "description"),
        ("BUILD", "", "description"),
    ];
    for (commit_type, scope, description) in test_cases {
        let result = validate(
            commit_types.clone(),
            true,
            commit_type,
            scope,
            description,
        );
        
        assert!(result.is_ok(), "Error: {}", result.unwrap_err());
        assert_eq!(result.unwrap(), true);
    }
}

```

This test _should_ be green for case insensitivity, but it ain't.